pkg/cloud/services/awsnode/cni.go

```diff
@@ -272,22 +271,23 @@ func (s *Service) deleteResource(ctx context.Context, remoteClient client.Client
 			return fmt.Errorf("deleting resource %s: %w", key, err)
 		}
 		s.scope.Debug(fmt.Sprintf("resource %s was not found, no action", key))
-	} else {
-		// resource found, delete if no label or not managed by helm
-		if val, ok := obj.GetLabels()[konfig.ManagedbyLabelKey]; !ok || val != "Helm" {
-			if err := remoteClient.Delete(ctx, obj, &client.DeleteOptions{}); err != nil {
-				if !apierrors.IsNotFound(err) {
-					return fmt.Errorf("deleting %s: %w", key, err)
-				}
-				s.scope.Debug(fmt.Sprintf(
-					"resource %s was not found, not deleted", key))
-			} else {
-				s.scope.Debug(fmt.Sprintf("resource %s was deleted", key))
-			}
-		} else {
-			s.scope.Debug(fmt.Sprintf("resource %s is managed by helm, not deleted", key))
+		return nil
+	}
+	// Check if the "prevent-deletion" label is present and set to "true", skip if present
```

Comment:
```suggestion
	// Don't delete if the `prevent-deletion` label exists. It could be there because CAPA added it (see below),
	// or because it was added externally, for example if a custom version of AWS CNI was already installed.
	// Either way, CAPA should not delete such a labelled CNI installation.
```